### DISCUSSION TITLE
How to log all url changes?

### DISCUSSION THREAD

**Discussion by Toshinaki** [remix-run/react-router#10676] [Q&A] [answered]
For debug purpose, I'd like to log current path and redirecting path when url changes.

For example:
```
curr: /auth/login
navigating: /path/after/login
...
```
How to implement this?

**Comment by brophdawg11** [comment, marked as answer]
The simplest way is to setup a `useEffect` in your root component based on `useLocation` but that can only log each new location.

If you want more advanced logic, if you're using a `RouterProvider` you can get more information if you leverage `useNavigation` as well to log in-flight navigations.
